src/ext/php7/serializer.c

```diff
@@ -313,14 +297,50 @@ void ddtrace_serialize_span_to_array(ddtrace_span_t *span, zval *array TSRMLS_DC
     add_assoc_long(el, "start", span->start);
     add_assoc_long(el, "duration", span->duration);
 
-    ADD_ELEMENT_IF_NOT_NULL("name");
-    ADD_ELEMENT_IF_NOT_NULL("resource");
-    ADD_ELEMENT_IF_NOT_NULL("service");
-    ADD_ELEMENT_IF_NOT_NULL("type");
+    // SpanData::$name defaults to fully qualified called name (set at span close)
+    zval *prop_name = ddtrace_spandata_property_name(span->span_data);
+    zval prop_name_as_string;
+    if (Z_TYPE_P(prop_name) != IS_NULL) {
+        ddtrace_convert_to_string(&prop_name_as_string, prop_name);
+        _add_assoc_zval_copy(el, "name", &prop_name_as_string);
+    }
+
+    // SpanData::$resource defaults to SpanData::$name
+    zval *prop_resource = ddtrace_spandata_property_resource(span->span_data);
+    zval prop_resource_as_string;
+    if (Z_TYPE_P(prop_resource) != IS_NULL) {
+        ddtrace_convert_to_string(&prop_resource_as_string, prop_resource);
+        _add_assoc_zval_copy(el, "resource", &prop_resource_as_string);
+        zval_dtor(&prop_resource_as_string);
```

Comment:
Let's extract a static function that basically does:
```
        ddtrace_convert_to_string(&prop_resource_as_string, prop_resource);
        _add_assoc_zval_copy(el, "resource", &prop_resource_as_string);
        zval_dtor(&prop_resource_as_string);
```
This way we can replace 4 lines with 1 for resource, service, and type.